api/src/convert/dcap_evidence.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+//! Convert to/from external::DcapEvidence
+
+use crate::{external, ConversionError};
+use mc_attest_verifier_types::DcapEvidence;
+use mc_sgx_dcap_types::{Collateral, Quote3};
+
+impl From<&DcapEvidence> for external::DcapEvidence {
+    fn from(src: &DcapEvidence) -> Self {
+        let mut dst = Self::new();
+
+        let quote = mc_util_serial::serialize(&src.quote).expect("Invalid Dcap Evidence");
+        dst.set_quote(quote);
+        let collateral = mc_util_serial::serialize(&src.collateral).expect("Invalid Dcap Evidence");
+        dst.set_collateral(collateral);
+        if let Some(report_data) = &src.report_data {
+            dst.set_enclave_report_data_contents(report_data.into());
+        }
+        dst
+    }
+}
+
+impl TryFrom<&external::DcapEvidence> for DcapEvidence {
+    type Error = ConversionError;
+    fn try_from(src: &external::DcapEvidence) -> Result<Self, Self::Error> {
+        let quote: Option<Quote3<Vec<u8>>> = mc_util_serial::deserialize(src.get_quote())
+            .map_err(|_| ConversionError::InvalidContents)?;
+        if quote.is_none() {
+            return Err(ConversionError::MissingField("quote".to_string()));
+        }
```

Comment:
:question: I'm curious why we have `prost` implementations and these `protobuf` ones.
The only reason that `quote` is an option on `DcapEvidence` is to appease the prost derive. It seems like it would be better to use `external::DcapEvidence` everwhere that we want to serialize `DcapEvidence` and then use this `TryFrom` to fail for missing fields. As it currently stands one could construct and pass around a `DcapEvidence` without a `quote` which should never happen in a well behaving system.

Reply:
Can't `quote` be made required in the prost derive and then it will no longer be an option? 

And on whether why we have both `prost` and `protobuf` - I think we use `prost` for protobuf messages we want to use inside the enclave (since `prost` is no_std compatible), but in some cases we also want to use the message in the GRPC api (`grpcio` stuff), which currently requires `protobuf` (they have a `prost` feature but we haven't tried switching to it).

Reply:
You mean the issue is a missing `Default` implementation?

Reply:
> You mean the issue is a missing `Default` implementation?
Yes sorry, partial thought in the original comment.
Prost uses `Self::default()` because everything is optional in protobufs so when an item is missing you need some kind of default either the `Option<T>` where None is a default or some kind of default value.

I may be trying to leverage the type system too much, but I feel if we allow creating a `DcapEvidence` without a structurally sound quote, collateral, and report data, then we force all consumers to both check for the evidence being in a valid format as well as having to verify (the signatures). It seems like we should limit when and where potentially "corrupt" data is passed around.

Reply:
I think consumers are going to have to check that anyway since there's no guarantees about anything inside a protobuf message...